src/BuilderDiscoverer.ts

```diff
@@ -19,7 +21,10 @@ export default class BuilderDiscoverer {
   }
 
   public discover(): Builders {
-    return this._discoverRecursively(this.cwd);
+    const packagesPaths = this._detectPaths();
```

Comment:
Lets rename for clarity: `packagePaths` -> `packageRootPaths`, `_detectPaths` -> `_detectRootPaths`. Because what you are really doing is detecting root paths and from them you scan for configs recursively